specs/jsonschema-core.md

```diff
@@ -415,20 +427,40 @@ keywords MUST NOT begin with this prefix.
 Implementations MUST refuse to evaluate schemas which contain keywords which
 they do not know how to process or explicitly choose not to process.
 
-## Keyword Behaviors
+## Keyword Behaviors {#keyword-behaviors}
 
-JSON Schema keywords fall into several general behavior categories. Assertions
-validate that an instance satisfies constraints, producing a boolean result.
-Annotations attach information that applications may use in any way they see
-fit. Applicators apply subschemas to parts of the instance and combine their
-results.
+JSON Schema keywords may exhibit one or more behaviors. This specification
+defines three such behaviors[^7]:
+
+- Assertions validate that an instance satisfies constraints, producing a
+  boolean result: `true` if the constraints are satisfied; `false` otherwise.
+- Annotations attach information to instance locations that applications may use
+  in any way they see fit.
+- Applicators apply subschemas to parts of the instance and combine their
+  results.
+
+[^7]: This specification also defines several operational directive keywords,
+such as `$id` and `$schema`, which do not exhibit these behaviors. Instead,
+these keywords provide metadata that instruct implementations on how to
+interpret and process the schema.
 
-Extension keywords SHOULD stay within these categories, keeping in mind that
+Extension keywords SHOULD be defined using these behaviors, keeping in mind that
 annotations in particular are extremely flexible. Complex behavior is usually
 better delegated to applications on the basis of annotation data than
 implemented directly as schema keywords. However, extension keywords MAY define
 other behaviors for specialized purposes.
 
+Implementations SHOULD NOT add unspecified behaviors to keywords.
```

Comment:
I'm not sure what this is aiming to prevent.

Behaviors outside the three above? I'm not sure what that would be. That also seems more the concern of the definition of an extension keyword rather than an implementation.

Or behaviors within the three, but that aren't in a specification? If so, that seems just the nature of a specification, that an implementation of it sticks to its specified behaviors.

Reply:
See, [this](https://github.com/json-schema-org/json-schema-spec/pull/1577#discussion_r2040716835) thread.

Reply:
See [this discussion](https://github.com/json-schema-org/json-schema-spec/pull/1577#discussion_r2040716835).